src/Http/Http.Extensions/gen/Microsoft.AspNetCore.Http.Generators.csproj

```diff
@@ -20,9 +20,13 @@
   <ItemGroup>
     <InternalsVisibleTo Include="Microsoft.AspNetCore.Http.Extensions.Tests" />
   </ItemGroup>
-  
+
   <ItemGroup>
     <Compile Include="$(SharedSourceRoot)IsExternalInit.cs" LinkBase="Shared" />
+    <Compile Include="$(SharedSourceRoot)RoslynUtils\BoundedCacheWithFactory.cs" LinkBase="Shared" />
+    <Compile Include="$(SharedSourceRoot)RoslynUtils\CodeWriter.cs" LinkBase="Shared" />
+    <Compile Include="$(SharedSourceRoot)RoslynUtils\WellKnownTypes.cs" LinkBase="Shared" />
```

Comment:
I think sharing code is ok for now, but doing this results in two caches. I think we should investigate whether we can have a shared library project output to the analyzers directory that generators, analyzers, and code fixers can reference.

For route pattern parsing, there is a lot more code, and we don't want to duplicate parsing and caching route patterns twice (once in analyzer assembly, once in generators assembly)

Reply:
> I think sharing code is ok for now, but doing this results in two caches. I think we should investigate whether we can have a shared library project output to the analyzers directory that generators, analyzers, and code fixers can reference.

This is an interesting proposal. @chsienki do you happen to know if there is a pattern for sharing a symbol cache across analyzers/generators in the same compilation process?

> For route pattern parsing, there is a lot more code, and we don't want to duplicate parsing and caching route patterns twice (once in analyzer assembly, once in generators assembly)

Roger that! I captured that in https://github.com/dotnet/aspnetcore/issues/46342 so we can start discussing what that might look like.